src/lib/sfge/BaseSfgeEngine.ts

```diff
@@ -243,25 +229,67 @@ export class SfgeEngine extends AbstractRuleEngine {
 				violations: []
 			};
 
-			result.violations.push({
-				ruleName: sfgeViolation.ruleName,
-				severity: sfgeViolation.severity,
-				message: sfgeViolation.message,
-				category: sfgeViolation.category,
-				url: sfgeViolation.url,
-				sinkLine: sfgeViolation.sinkLineNumber || null,
-				sinkColumn: sfgeViolation.sinkColumnNumber || null,
-				sinkFileName: sfgeViolation.sinkFileName || "",
-				sourceLine: sfgeViolation.sourceLineNumber,
-				sourceColumn: sfgeViolation.sourceColumnNumber,
-				sourceType: sfgeViolation.sourceType,
-				sourceMethodName: sfgeViolation.sourceVertexName
-			});
+			result.violations.push(this.convertViolation(sfgeViolation));
 			resultMap.set(indexFile, result);
 		}
 		return [...resultMap.values()];
 	}
 
+	protected abstract convertViolation(sfgeViolation: SfgeViolation): RuleViolation;
```

Comment:
The subvariants have different ways of converting an SFGE violation into a real violation, so this is abstract.